doc/userguide/firewall/firewall-design.rst

```diff
@@ -0,0 +1,182 @@
+Firewall Mode Design
+====================
+
+.. note:: In Suricata 8 the firewall mode is experimental and subject to change.
+
+The firewall mode in Suricata allows the use of a ruleset that has different
+properties than the default "threat detection" rulesets:
+
+1. default policy is ``drop``, meaning a firewall ruleset needs to specify what
+   is allowed
+2. firewall rules are loaded from separate files
+3. firewall rules use a new action ``accept``
+4. firewall rules are required to use explicit action scopes and rule hooks (see below)
+5. evaluation order is as rules are in the file(s), per protocol state
+
+Concepts
+--------
+
+* ``table`` collection of rules with different properties: ``packet:filter``, ``packet:td``,
+  ``app:<proto>:<state>``, ``app:td``. These are built-in. No custom tables can be created.
+* ``state`` controls a specific protocol state at which a rule is evaluated. Examples are
+  ``tcp.flow_start`` or ``tls.client_body_done``.
+
+The tables for the application layer are per app layer protocol and per protocol state.
+For example, ``app:http:request_line``.
+
+
+Actions and Action Scopes
+-------------------------
+
+Firewall rules require action scopes to be explicitly specified.
+
+accept
+~~~~~~
+
+``accept`` is used to issue an accept verdict to the packet, flow or hook.
+
+* ``packet`` accept this packet
+* ``flow`` accept the rest of the packets in this flow
+* ``hook`` accept rules for the current hook/state, evaluate the next tables
```

Comment:
Is `state` a better name for this? The state of a connection or protocol makes sense and is in-line with some current lingo.  Plus below we describe hook as state.. For example: `Explicit rule hook (states)`.